worktree.go

```diff
@@ -713,30 +713,65 @@ func (w *Worktree) readGitmodulesFile() (*config.Modules, error) {
 }
 
 // Clean the worktree by removing untracked files.
+// An empty dir could be removed - this is what  `git clean -f -d .` does.
 func (w *Worktree) Clean(opts *CleanOptions) error {
 	s, err := w.Status()
 	if err != nil {
 		return err
 	}
 
-	// Check Worktree status to be Untracked, obtain absolute path and delete.
-	for relativePath, status := range s {
-		// Check if the path contains a directory and if Dir options is false,
-		// skip the path.
-		if relativePath != filepath.Base(relativePath) && !opts.Dir {
-			continue
+	var clean func(string, []os.FileInfo) error
+	clean = func(dir string, files []os.FileInfo) error {
+		for _, fi := range files {
+			if fi.Name() == ".git" {
+				continue
+			}
+
+			// relative path under the root
+			path := filepath.Join(dir, fi.Name())
+			if fi.IsDir() {
+				if !opts.Dir {
+					continue
+				}
+
+				subfiles, err := w.Filesystem.ReadDir(path)
+				if err != nil {
+					return err
+				}
+				err = clean(path, subfiles)
+				if err != nil {
+					return err
+				}
+			} else {
+				// check if file is 'Untracked'
+				stat, ok := (s)[filepath.ToSlash(path)]
+				if ok && stat.Worktree == Untracked {
+					if err = w.Filesystem.Remove(path); err != nil {
+						return err
+					}
+				}
+			}
 		}
 
-		// Remove the file only if it's an untracked file.
-		if status.Worktree == Untracked {
-			absPath := filepath.Join(w.Filesystem.Root(), relativePath)
-			if err := os.Remove(absPath); err != nil {
+		if opts.Dir {
```

Comment:
also will be nice move this code to a function something like `doCleanDirectories`